src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -115,12 +116,16 @@ func (r *StatefulSetResource) runUpdate(
 		return err
 	}
 
-	// if !ALL_PODS_ARE_UPDATED() {
-	// 	return &RequeueAfterError{
-	// 		RequeueAfter: wait.Jitter(r.decommissionWaitInterval, decommissionWaitJitterFactor),
-	// 		Msg:          "waiting for all pods to be updated",
-	// 	}
-	// }
+	ok, err := r.areAllPodsUpdated(ctx)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return &RequeueAfterError{
+			RequeueAfter: wait.Jitter(r.decommissionWaitInterval, decommissionWaitJitterFactor),
+			Msg:          "waiting for all pods to be updated",
+		}
+	}
```

Comment:
Looking at the git history I didn't see `ClusterUpdatePodCondition` being set to `false`

Reply:
When the pod is deleted, the new pod won't have that condition.